app/javascript/Pages/Home.jsx

```diff
@@ -1,78 +1,132 @@
 import React, { Component } from "react";
 import Navbar from "./components/Navbar/Navbar";
 import Footer from "./components/Footer/Footer";
-import { Head, Link } from "@inertiajs/inertia-react";
-import "../stylesheets/Home.css";
+import { Head } from "@inertiajs/inertia-react";
 import { useState, useEffect } from "react";
 import SocialCard from "./components/SocialCard";
+import "../stylesheets/home.scss";
+import ButtonDifficultyFilter from "./components/ButtonDifficultyFilter";
+import CardCategory from "./components/CardCategory";
+import CardPopular from "./components/CardPopular";
 
 function Homepage(props) {
   const token = document.querySelector("[name=csrf-token]").content;
   const headers = { "X-CSRF-Token": token };
 
-  const [items, setUsers] = useState([]);
+  const [items, setItems] = useState([]);
+  const [cats, setCat] = useState([]);
+  const [pclasses, setPClasses] = useState([]);
+
+  const fetchClassesDifficulty = async (difficulty) => {
+    let newClassData;
+    try {
+      const response = await fetch(`/api/classes?difficulty=${difficulty}`);
```

Comment:
**issue (blocking)**: fetch is not supported in every browser (https://caniuse.com/fetch). We usually like to use axios.
Also, instead of using a try catch (which should only be used as a last resort) you can use the .catch chaining of fetch or axios: `axios.get('url').catch((e) => { console.log(e) })`